src/poetry/core/packages/dependency_group.py

```diff
@@ -3,30 +3,38 @@
 from collections import defaultdict
 from typing import TYPE_CHECKING
 
+from packaging.utils import NormalizedName
+from packaging.utils import canonicalize_name
+
 from poetry.core.version.markers import parse_marker
 
 
 if TYPE_CHECKING:
     from poetry.core.packages.dependency import Dependency
     from poetry.core.version.markers import BaseMarker
 
-MAIN_GROUP = "main"
+MAIN_GROUP: NormalizedName = canonicalize_name("main")
 
 
 class DependencyGroup:
     def __init__(
         self, name: str, *, optional: bool = False, mixed_dynamic: bool = False
     ) -> None:
-        self._name: str = name
+        self._name: NormalizedName = canonicalize_name(name)
+        self._pretty_name: str = name
         self._optional: bool = optional
         self._mixed_dynamic = mixed_dynamic
         self._dependencies: list[Dependency] = []
```

Comment:
Are `Dependency` names already normalized? I see fns like `remove_dependency` assume they are.

Reply:
`Dependency` inherits from `PackageSpecification`, which has a normalized `name` and an unnormalized `pretty_name`.